tests/bluetooth/tester/src/btp_vcp.c

```diff
@@ -0,0 +1,587 @@
+}
+
+void aics_desc(uint8_t *data)
+{
+	const struct aics_audio_desc *cmd = (void *) data;
+
+	LOG_DBG("AICS description");
+
+	if (cmd->desc_len > BT_AICS_MAX_INPUT_DESCRIPTION_SIZE) {
+		LOG_ERR("Too long input (%u chars required)",
+			BT_AICS_MAX_INPUT_DESCRIPTION_SIZE);
+		goto rsp;
+	}
+
+	for (int i = 0; i < CONFIG_BT_VCP_VOL_REND_AICS_INSTANCE_COUNT; i++) {
+		int err = bt_aics_description_set(included.aics[i], cmd->desc);
+
+		if (err) {
+			tester_rsp(BTP_SERVICE_ID_AICS, AICS_DESCRIPTION,
+				   CONTROLLER_INDEX, BTP_STATUS_FAILED);
+			return;
+		}
+	}
+rsp:
+	tester_rsp(BTP_SERVICE_ID_AICS, AICS_DESCRIPTION, CONTROLLER_INDEX,
+		   BTP_STATUS_FAILED);
+}
+
+void tester_handle_aics(uint8_t opcode, uint8_t index, uint8_t *data, uint16_t len)
+{
+	switch (opcode) {
+	case AICS_READ_SUPPORTED_COMMANDS:
+		aics_supported_commands(data, len);
+		break;
+	case AICS_SET_GAIN:
+		aics_set_gain(data);
+		break;
+	case AICS_MUTE:
+		aics_mute();
+		break;
+	case AICS_UNMUTE:
+		aics_unmute();
+		break;
+	case AICS_MAN_GAIN:
+		aics_man_gain();
+		break;
+	case AICS_AUTO_GAIN:
+		aics_auto_gain();
+		break;
+	case AICS_MAN_GAIN_ONLY:
+		aics_auto_gain_only();
+		break;
+	case AICS_AUTO_GAIN_ONLY:
+		aics_auto_gain_only();
+		break;
+	case AICS_DESCRIPTION:
+		aics_desc(data);
+		break;
+	case AICS_MUTE_DISABLE:
+		aics_mute_disable();
+		break;
+	default:
+		tester_rsp(BTP_SERVICE_ID_AICS, opcode, index,
+			   BTP_STATUS_UNKNOWN_CMD);
+		break;
+	}
+}
+
+/* Volume Offset Control Service */
+static void vocs_supported_commands(void)
+{
+	struct net_buf_simple *buf = NET_BUF_SIMPLE(BTP_DATA_MAX_SIZE);
+
+	net_buf_simple_init(buf, 0);
+	net_buf_simple_add_u8(buf, VOCS_READ_SUPPORTED_COMMANDS);
+	net_buf_simple_add_u8(buf, VOCS_UPDATE_LOC);
+	net_buf_simple_add_u8(buf, VOCS_UPDATE_DESC);
+
+	tester_send(BTP_SERVICE_ID_VOCS, VOCS_READ_SUPPORTED_COMMANDS,
+		    CONTROLLER_INDEX, buf->data, buf->len);
+}
+
+static void vocs_state_cb(struct bt_vocs *inst, int err, int16_t offset)
+{
+	LOG_DBG("VOCS state callback");
+}
+
+static void vocs_location_cb(struct bt_vocs *inst, int err, uint32_t location)
+{
+	LOG_DBG("VOCS location callback");
+}
+
+static void vocs_description_cb(struct bt_vocs *inst, int err,
+				char *description)
+{
+	LOG_DBG("VOCS desctripion callback");
+}
+
+static struct bt_vocs_cb vocs_cb = {
+	.state = vocs_state_cb,
+	.location = vocs_location_cb,
+	.description = vocs_description_cb
+};
+
+void vocs_audio_desc(uint8_t *data)
+{
+	const struct vocs_audio_desc *cmd = (void *) data;
+	uint8_t desc_str[16];
+
+	LOG_DBG("VOCS description");
+
+	/* FIXME: Spec does not specify maximum desc length */
+	if (cmd->desc_len > sizeof(desc_str)) {
+		LOG_ERR("Too long input (%u chars required)", 16);
+		goto rsp;
+	}
+
+	strncpy(desc_str, cmd->desc, cmd->desc_len);
+
+	for (int i = 0; i < CONFIG_BT_VCP_VOL_REND_VOCS_INSTANCE_COUNT; i++) {
+		int err = bt_vocs_description_set(included.vocs[i], desc_str);
```

Comment:
This is different handling of the description that what we have in `aics_desc`, although it still does not ensure that `desc_str` is NULL terminated, so the `strncpy` doesn't really provide anything compared to just using `cmd->desc` as you are doing in `aics_desc`. 

Lastly, the AICS and VOCS descriptions are UTF-8 strings, and thus you should use use `utf8_lcpy` instead of `strncpy` if you really want to copy with a maximum length (`utf8_lcpy` also ensures that the resulting string is NULL terminated. 

Also the same comment w.r.t using `char` instead of `uint8_t` for string values.

Reply:
done